OWML.ModHelper.Input/ModInputHandler.cs

```diff
@@ -0,0 +1,440 @@
+                else
+                {
+                    return -1;
+                }
+                i++;
+            }
+            Array.Sort(curcom);
+            Int64 hash = 0;
+            for (i = 0; i < 7; i++)
+            {
+                hash = hash * _maxUsefulKey + curcom[i];
+            }
+            if (hash < _maxUsefulKey && _gameBindingCounter[hash] > 0)
+            {
+                return -3;
+            }
+            return (_comboRegistry.ContainsKey(hash) && !forRemoval ? -3 : hash);
+        }
+
+        public int RegisterCombination(IModInputCombination combination)
+        {
+            var castCombo = (ModInputCombination)combination;
+            castCombo.ClearSingles();
+            if (castCombo == null || castCombo.Combo == null)
+            {
+                _console.WriteLine("combination is null");
+                return -1;
+            }
+            string[] combs = castCombo.Combo.ToLower().Split('/');
+            List<Int64> combos = new List<Int64>();
+            foreach (string comstr in combs)
+            {
+                Int64 hash = ParseCombination(comstr);
+                if (hash <= 0)
+                {
+                    return (int)hash;
+                }
+                else
+                {
+                    combos.Add(hash);
+                }
+            }
+            foreach (Int64 comb in combos)
+            {
+                _comboRegistry.Add(comb, combination);
+                if (comb < _maxUsefulKey)
+                {
+                    castCombo.AddSingle((KeyCode)comb);
+                }
+            }
+            _logger.Log("succesfully registered " + castCombo.Combo);
+            return 1;
+        }
+
+        public int UnregisterCombination(IModInputCombination combination)
+        {
+            var castCombo = (ModInputCombination)combination;
+            if (_comboRegistry.ContainsValue(combination))
+            {
+                if (combination == null || castCombo.Combo == null)
+                {
+                    _console.WriteLine("combination is null");
+                    return -1;
+                }
+                string[] individualCombos = castCombo.Combo.ToLower().Split('/');
+                List<Int64> hashes = new List<Int64>();
+                foreach (string comboString in individualCombos)
+                {
+                    Int64 hash = ParseCombination(comboString, true);
+                    if (hash <= 0 && hash > -3)
+                    {
+                        return (int)hash;
+                    }
+                    else
+                    {
+                        hashes.Add(hash);
+                    }
+                }
+                foreach (Int64 comb in hashes)
+                {
+                    _comboRegistry.Remove(comb);
+                }
+                _logger.Log("succesfully unregistered " + castCombo.Combo);
+                return -3;
```

Comment:
What's with all of these seemingly random return values? -2, -3, 1, etc?

Reply:
added enum, not sure if place is correct (placed in IModInputHandler.cs)

Reply:
separate now